[PRO-1611] Pass namespace explicitly in headers when AUTH_PROTOCOL=none

Danten opened the pull request:
If AUTH_PROTOCOL=none then namespaceExtractor will now return the namespace in the header (x-ats-namespace) if it exists otherwise return the default one.

GENIVI webserver uses loggedIn.id as a namespace and sets the header on all request. Two endpoints that got the namespace via a queryparameter are now proxied via webserver to add the correct namespace.

simao commented:
You mean they were already proxied, but now they have explicit routes?

simao commented on common/src/main/scala/org/genivi/sota/http/NamespaceDirectives.scala:
```diff
@@ -10,16 +12,28 @@ import org.slf4j.LoggerFactory
-  lazy val defaultNamespaceExtractor: Directive1[Namespace] =
-    BasicDirectives.provide(configNamespace(ConfigFactory.load()))
+  lazy val defaultNamespaceExtractor: Directive1[Namespace] = {
+    extractRequest flatMap { req =>
```
we should prefer `.flatMap` http://docs.scala-lang.org/style/method-invocation.html

simao commented on common/src/main/scala/org/genivi/sota/http/NamespaceDirectives.scala:
```diff
@@ -10,16 +12,28 @@ import org.slf4j.LoggerFactory
+        case None => BasicDirectives.provide(configNamespace(ConfigFactory.load()))
+      }
+    }
+  }
```
with `Directives._` imported you should not need `BasicDirectives`

simao commented on web-server/app/org/genivi/webserver/controllers/Application.scala:
```diff
@@ -114,11 +114,26 @@ class Application @Inject() (ws: WSClient,
 
+  def apiProxyWithNs[T](apiUri: String): Action[RawBuffer]
+    = AsyncStack(parse.raw, AuthorityKey -> User) { implicit req =>
+      {
```
I think these braces are meaningless?(?)

Danten commented:
Yes they were proxied before, but I changed the javascript code so that it no longer provides the query-parameter for the namespace. So the namespace is added by the webserver which is why it is not a normal proxy.

Danten commented:
@simao I updated the code, and have tried it locally by adding a vehicle (which tests the /devices route) and added a package which tests the resolve route.
